0-bootstrap/sa.tf

```diff
@@ -67,6 +67,21 @@ locals {
       "roles/compute.xpnAdmin",
     ],
   }
+
+  granular_sa_seed_project = {
```

Comment:
Let's add a comment here this will allow individual stages to access state for all other stages (incase users want to restrict this)

Reply:
Good catch.

actually, these roles are redundant with the role `roles/storage.admin` granted in the bucket in the `tf_cloudbuild_workspace` [module](https://github.com/terraform-google-modules/terraform-google-bootstrap/blob/e838b1d381a2cf88a36bbc59110994d5d0df1fb9/modules/tf_cloudbuild_workspace/buckets.tf#L74-L78)
since we are using a single tf state bucket created in the seed project to host all the states like we did before using the workspace module.

to be able to revoke the access of the others service accounts to one of the state buckets, we will need to:

- stop using the tf state bucket from the seed project
- create a tf state bucket for each workspace in the CI/CD project, setting `var.create_state_bucket` to true
- grant read roles in a bucket for the other service accounts that need to read the outputs

@bharathkkb should we do this change?

Reply:
I see, I think we can keep this in the PR now and track it in a separate PR. My hope with the workspace module was in the future we can split this out for more granular permissions.